### PR TITLE
Run the migrator in the e2e test

### PR DESCRIPTION
**Pull request by caesarxuchao**
Add a simple case to the e2e test to actually exercise the migrator.

### CODE REVIEW

**Comment by caesarxuchao** [comment]
https://github.com/kubernetes/test-infra/pull/10410 should have fixed the docker daemon issue. Retry the test.

**Comment by caesarxuchao** [comment]
Still com

**Comment by caesarxuchao** [comment]
@krzyzacy the test is still failing with the "docker daemon not found" error. Can you help take a look? Thanks. (The test job config is [here](https://github.com/caesarxuchao/test-infra/blob/5eee6b2b9494cac8cd61063cf5e804e758163f5e/config/jobs/kubernetes-sigs/kube-storage-version-migrator/presubmits.yaml#L21))

File: test/e2e/test-cmd.sh [hunk 1]
```diff
@@ -32,19 +36,76 @@ function wait-for-apiserver()
...
+      echo ""
+      sleep 10
+    fi
+  done
+
+  echo " initializer logs:"
+  kubectl logs --namespace=kube-storage-migration -l job-name=initializer || true
+  echo "migrator logs:"
+  kubectl logs --namespace=kube-storage-migration -l app=migrator || true
+  return 1
```
Comment (yliaog) [inline test/e2e/test-cmd.sh]: why return 1?

File: test/e2e/test-cmd.sh [hunk 2]
```diff
@@ -32,19 +36,76 @@ function wait-for-apiserver()
...
-
-# Change the apiserver manifest to change the storage version.
-gcloud compute --project "${PROJECT}" ssh --zone "${KUBE_GCE_ZONE}" "${CLUSTER_NAME}-master" --command \
-"sudo sed -i \"s/--v=/--storage-versions=apps\/v1beta2 --v=/\" /etc/kubernetes/manifests/kube-apiserver.manifest"
+# This is to enable docker push. Running it here to fail early.
+gcloud auth configure-docker
 
-wait-for-apiserver 
+# TODO: create more objects.
+# STEP 1: create an object
```
Comment (yliaog) [inline test/e2e/test-cmd.sh]: so you don't start with creating a cluster?

File: test/e2e/test-cmd.sh (hunk 1 above)
Comment (caesarxuchao) [reply test/e2e/test-cmd.sh]: Good point. Printed to logs.

File: test/e2e/test-cmd.sh [hunk 3]
```diff
@@ -59,7 +120,35 @@ EOF
...
+
+pushd "${MIGRATORROOT}/manifests.local"
+kubectl apply -f namespace-rbac.yaml
+kubectl apply -f initializer.yaml
+kubectl apply -f migrator.yaml
+popd
+
+wait-for-migration
+
+# STEP 4: verify the object has been migrated.
```
[the remark this answers is not in the document]
Comment (caesarxuchao) [reply test/e2e/test-cmd.sh]: wait-for-migration is checking if all the migration CR have the "Succeeded" condition. Do you have other checks in mind?

File: test/e2e/test-cmd.sh [hunk 4]
```diff
@@ -59,7 +121,35 @@ EOF
 
 kubectl get controllerrevisions sample --namespace=default
 
-# TODO: run the migrator
+# STEP 2: change the apiserver configuration of --storage-versions.
+
+# Change the apiserver manifest to change the storage version.
+gcloud compute --project "${PROJECT}" ssh --zone "${KUBE_GCE_ZONE}" "${CLUSTER_NAME}-master" --command \
+"sudo sed -i \"s/--v=/--storage-versions=apps\/v1beta2 --v=/\" /etc/kubernetes/manifests/kube-apiserver.manifest"
```
Comment (yliaog) [inline test/e2e/test-cmd.sh]: so the k8s needs to support apps/v1beta2, did you fix the k8s release in this test?

Comment (caesarxuchao) [reply test/e2e/test-cmd.sh]: Good point, we'll need to update the config [here](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/kube-storage-version-migrator/presubmits.yaml#L30) to pin the k8s version. On the other hand, I want to test the migrator with the latest Kubernetes to ensure compatibility. In the future, we can check the apiserver version in this test script, or switch to a more dynamic test setup. I don't know which path we'll take yet, so I left a TODO.

**Pull request merged**